src/server/game/Spells/Auras/SpellAuras.cpp

```diff
@@ -2039,6 +2046,34 @@ void Aura::TriggerProcOnEvent(uint8 procEffectMask, AuraApplication* aurApp, Pro
         Remove();
 }
 
+void Aura::HeartbeatResistance(uint32 diff, Unit* caster)
+{
+    Unit* target = GetOwner()->ToUnit();
+
+    if (!target || !caster)
+        return;
+
+    if (m_heartBeatTimer <= diff)
+    {
+        if (target->GetTypeId() == TYPEID_UNIT && caster->GetTypeId() == TYPEID_PLAYER)
+        {
+            SpellSchoolMask schoolMask = m_spellInfo->GetSchoolMask();
+            uint32 resistance = schoolMask != SPELL_SCHOOL_MASK_NORMAL ? target->GetResistance(GetFirstSchoolInMask(schoolMask)) : 0;
+            uint32 breakPct = uint32(resistance / powf(float(target->getLevel()), 1.441f) * 0.10f * 100) + 5;
+
+            if (roll_chance_i(breakPct))
+            {
+                Remove();
+                TC_LOG_DEBUG("spells", "Aura::HeartbeatResistance: Breaking creature aura %u. Seconds passed %u with chance %u.", m_spellInfo->Id, m_heartBeatTimer, breakPct);
```

Comment:
Log not needed and can hamper performance

Reply:
no debug logs are "needed", and we have a ton of them. this is fine.

Reply:
We really need a general consensus. You could as easily debug with breakpoints